src/Compilers/CSharp/Portable/Symbols/FieldSymbol.cs

```diff
@@ -424,7 +425,7 @@ public virtual FieldSymbol CorrespondingTupleField
         {
             get
             {
-                return null;
+                return (ContainingType.IsDefinition && TupleElementIndex >= 0) ? this : null;
```

Comment:
>ContainingType.IsDefinition [](start = 24, length = 27)

`TupleElementIndex` includes this check.

Reply:
That property is `virtual` though. Do we feel comfortable depending on all the overrides also enforcing this behavior?